[Lens] Hide column in table

flash1293 opened the pull request:
Fixes https://github.com/elastic/kibana/issues/85140

This PR allows to hide a column in lens data table. There's an action in the cell popover as well as a switch in the dimension editor:
<img width="450" alt="Screenshot 2021-02-03 at 14 21 13" src="https://user-images.githubusercontent.com/1508364/106752828-2bd5f080-662b-11eb-914e-ff0e83c73824.png">
<img width="399" alt="Screenshot 2021-02-03 at 14 21 08" src="https://user-images.githubusercontent.com/1508364/106752831-2e384a80-662b-11eb-8f16-0cd18c99bc82.png">

This PR also restructures the data table shape a little on all levels to make it easier to work with (especially considering the fact we will add many more settings to individual columns)

# New state and expression shape

The old state looked like this:
```
{
  layers: Array<{
    columnIds: string[];
    columnWidth: ColumnWidthState[];
    layerId: string;
  }>;
  sorting?: SortingState;
}
```

As we are going to add many more config settings to specific columns, this PR changes the state to this easier to manager shape:
```
{
  layerId: string;
  columns: Array<ColumnState>;
  sorting?: SortingState;
}

interface ColumnState {
  columnId: string;
  width?: number;
  hidden?: boolean;
  // to be continued
}
```

This is also reflected on the expression layer, the new layout looks like this:
```
lens_datatable column={lens_datatable_column columnId="a" width=300} column={lens_datatable_column columnId="b"} ...
```

### Functional tests

I slightly restructured functional tests by starting a separate "table" test file, moving some things over from the very crowded "smokescreen" test file, extending it with a test for hiding the column.

flash1293 commented:
@elasticmachine merge upstream

flash1293 commented:
Jenkins, test this

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/)
* Commit: e821ca608ce72f1f7e71887100b6ec36bb7e62c9
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/execution/node/618/log)
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/execution/node/751/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/)
* Commit: 82d7e9f9ce1354bdaceea075591c9b3786b59159

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/a4c884b92bdc081242e3f520f051222f35b9b6b9) | [after](https://github.com/elastic/kibana/commit/82d7e9f9ce1354bdaceea075591c9b3786b59159) | diff |
| --- | --- | --- | --- |
| `lens` | 465 | 475 | +10 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a4c884b92bdc081242e3f520f051222f35b9b6b9) | [after](https://github.com/elastic/kibana/commit/82d7e9f9ce1354bdaceea075591c9b3786b59159) | diff |
| --- | --- | --- | --- |
| `lens` | 851.7KB | 860.2KB | +8.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/a4c884b92bdc081242e3f520f051222f35b9b6b9) | [after](https://github.com/elastic/kibana/commit/82d7e9f9ce1354bdaceea075591c9b3786b59159) | diff |
| --- | --- | --- | --- |
| `lens` | 34.4KB | 34.4KB | -3.0B |

### History
* :broken_heart: [Build #101084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/) failed e821ca608ce72f1f7e71887100b6ec36bb7e62c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/)
* Commit: c7240abe135cda0ea68b10cd460e25cfac2b0fac

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/c7240abe135cda0ea68b10cd460e25cfac2b0fac) | diff |
| --- | --- | --- | --- |
| `lens` | 465 | 475 | +10 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/c7240abe135cda0ea68b10cd460e25cfac2b0fac) | diff |
| --- | --- | --- | --- |
| `lens` | 851.9KB | 860.3KB | +8.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/c7240abe135cda0ea68b10cd460e25cfac2b0fac) | diff |
| --- | --- | --- | --- |
| `lens` | 34.4KB | 34.4KB | -3.0B |

### History
* :green_heart: [Build #101142](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/) succeeded 82d7e9f9ce1354bdaceea075591c9b3786b59159
* :broken_heart: [Build #101084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/) failed e821ca608ce72f1f7e71887100b6ec36bb7e62c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/)
* Commit: 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/da1a4e947a6c49d82adf1cc66d26c376a54e4bd0) | [after](https://github.com/elastic/kibana/commit/00b7ab667612cfa09e5db3a707c3bf74eab1e6d5) | diff |
| --- | --- | --- | --- |
| `lens` | 474 | 475 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/da1a4e947a6c49d82adf1cc66d26c376a54e4bd0) | [after](https://github.com/elastic/kibana/commit/00b7ab667612cfa09e5db3a707c3bf74eab1e6d5) | diff |
| --- | --- | --- | --- |
| `lens` | 859.3KB | 861.7KB | +2.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/da1a4e947a6c49d82adf1cc66d26c376a54e4bd0) | [after](https://github.com/elastic/kibana/commit/00b7ab667612cfa09e5db3a707c3bf74eab1e6d5) | diff |
| --- | --- | --- | --- |
| `lens` | 34.3KB | 34.2KB | -100.0B |

### History
* :green_heart: [Build #101422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/) succeeded c7240abe135cda0ea68b10cd460e25cfac2b0fac
* :green_heart: [Build #101142](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/) succeeded 82d7e9f9ce1354bdaceea075591c9b3786b59159
* :broken_heart: [Build #101084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/) failed e821ca608ce72f1f7e71887100b6ec36bb7e62c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

wylieconlon commented on x-pack/plugins/lens/public/datatable_visualization/components/columns.tsx:
```diff
@@ -173,6 +173,16 @@ export const createGridColumns = (
+              {
+                color: 'text',
+                size: 'xs',
+                onClick: () => onColumnHide({ columnId: field }),
```
It seems like this does not update the visibility indicators in the main menu: when I use the `Hide` button from the column menu, not from the visualization menu, I get inconsistent results:

<img width="876" alt="Screen Shot 2021-01-29 at 5 41 40 PM" src="https://user-images.githubusercontent.com/666475/106334759-5144b200-6259-11eb-8ea7-03266e397381.png">

wylieconlon commented on x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx:
```diff
@@ -120,11 +143,15 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
+      columnConfig.columns
+        .filter((col) => !!col.columnId && !col.hidden)
+        .map((col) => col.columnId),
+    [columnConfig]
```
Maybe there's a JS deep equality issue that causes visibleColumns to get stale when triggered from the visualization?

wylieconlon commented on x-pack/test/functional/page_objects/lens_page.ts:
```diff
@@ -561,6 +561,16 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
+      await testSubjects.setEuiSwitch(id, isChecked ? 'uncheck' : 'check');
+      await testSubjects.click('lnsColumnsButton');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+    },
```
It looks like this is toggling from the menu containing all of the columns, but it would be great if we could also interact with the datagrid header columns directly- this would help test the bug I found.

dej611 reviewed (changes requested):
I've encountered some issues on the hide behaviour.
I suspect most of these issues are related to the fact that columns configuration are computed over `columnsConfig` rather than `visibleColumns` -  just a guess.

* First `hide` runs twice reverting the hide somehow:
  
  ![hidden_columns_datatable](https://user-images.githubusercontent.com/924948/106441378-4452d900-647a-11eb-8901-a0dd4a2cc55d.gif)

* If the `Hide` commands comes from the column dropdown, it goes out of sync with the Column visibility dropdown menu
  
  <img width="834" alt="Screenshot 2021-02-01 at 10 43 46" src="https://user-images.githubusercontent.com/924948/106441565-7e23df80-647a-11eb-850e-8eb7f5e70f87.png">

* If the column hidden has a `sorting` criteria configured, the `Hide` command only removes the criteria keeping the visibility
  
![hidden_columns_datatable_sorting](https://user-images.githubusercontent.com/924948/106441859-db1f9580-647a-11eb-984f-38e3dad5c55b.gif)

* If a hidden column has a custom size, resizing another column brings back the hidden column

![hidden_columns_datatable_resizing](https://user-images.githubusercontent.com/924948/106442777-fa6af280-647b-11eb-92ea-313ef524f0e4.gif)

* Toggling the column visibility from the dropdown menu triggers a waterfall of `RangerError` for max call stack size in the console
  
<img width="422" alt="Screenshot 2021-02-01 at 10 48 16" src="https://user-images.githubusercontent.com/924948/106442031-1752f600-647b-11eb-8e0c-f598f180738e.png">

* I'd suggest to make the last visible column visibility disabled. I cannot see no use for it, at least in a single visualization configuration. Maybe in a dashboard could have "some sense" with some sort of filters, but that's a different flow.

<img width="833" alt="Screenshot 2021-02-01 at 10 49 44" src="https://user-images.githubusercontent.com/924948/106442480-9ba57900-647b-11eb-91e3-26c1558754b6.png">

flash1293 commented:
Thanks for the review everyone. I think all of the reported problems were caused by this pretty basic mistake:
```
+                hidden: !column.hidden,
-                hidden: !!column.hidden,
```

The edit action didn't actually toggle the hidden flag... It should be fixed now

flash1293 commented:
> Toggling the column visibility from the dropdown menu triggers a waterfall of RangerError for max call stack size in the console

I saw this behavior before, AFAIK it's a bug of the data table. I will check with the most recent update whether it still persists.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/)
* Commit: 064794b7e707dd21cc192839a2de5457ec40df44

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/84d49f11238c76c806b360a28f1d579dde38ab16) | [after](https://github.com/elastic/kibana/commit/064794b7e707dd21cc192839a2de5457ec40df44) | diff |
| --- | --- | --- | --- |
| `lens` | 474 | 475 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/84d49f11238c76c806b360a28f1d579dde38ab16) | [after](https://github.com/elastic/kibana/commit/064794b7e707dd21cc192839a2de5457ec40df44) | diff |
| --- | --- | --- | --- |
| `lens` | 859.3KB | 861.6KB | +2.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/84d49f11238c76c806b360a28f1d579dde38ab16) | [after](https://github.com/elastic/kibana/commit/064794b7e707dd21cc192839a2de5457ec40df44) | diff |
| --- | --- | --- | --- |
| `lens` | 34.3KB | 34.2KB | -100.0B |

### History
* :green_heart: [Build #102128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/) succeeded 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5
* :green_heart: [Build #101422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/) succeeded c7240abe135cda0ea68b10cd460e25cfac2b0fac
* :green_heart: [Build #101142](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/) succeeded 82d7e9f9ce1354bdaceea075591c9b3786b59159
* :broken_heart: [Build #101084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/) failed e821ca608ce72f1f7e71887100b6ec36bb7e62c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/)
* Commit: db8827776ea554aa0de15fc9c3796b8e605afa6a
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup9](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/execution/node/736/log)
* [Execute kibana-ciGroup2](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/execution/node/708/log)

### History
* :green_heart: [Build #102525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/) succeeded 064794b7e707dd21cc192839a2de5457ec40df44
* :green_heart: [Build #102128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/) succeeded 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5
* :green_heart: [Build #101422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/) succeeded c7240abe135cda0ea68b10cd460e25cfac2b0fac
* :green_heart: [Build #101142](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/) succeeded 82d7e9f9ce1354bdaceea075591c9b3786b59159
* :broken_heart: [Build #101084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/) failed e821ca608ce72f1f7e71887100b6ec36bb7e62c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101084/","number":"101084","commit":"e821ca608ce72f1f7e71887100b6ec36bb7e62c9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/","number":"102906","commit":"db8827776ea554aa0de15fc9c3796b8e605afa6a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/)
* Commit: db3ab50aeb27af42cda55f50e8667e11805059b0
* [Documentation Changes](https://kibana_88680.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e73b3778ed0294cb3f72d14486c29988cd8ec20d) | [after](https://github.com/elastic/kibana/commit/db3ab50aeb27af42cda55f50e8667e11805059b0) | diff |
| --- | --- | --- | --- |
| `lens` | 478 | 479 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e73b3778ed0294cb3f72d14486c29988cd8ec20d) | [after](https://github.com/elastic/kibana/commit/db3ab50aeb27af42cda55f50e8667e11805059b0) | diff |
| --- | --- | --- | --- |
| `lens` | 873.7KB | 876.0KB | +2.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e73b3778ed0294cb3f72d14486c29988cd8ec20d) | [after](https://github.com/elastic/kibana/commit/db3ab50aeb27af42cda55f50e8667e11805059b0) | diff |
| --- | --- | --- | --- |
| `lens` | 34.3KB | 34.2KB | -100.0B |

### History
* :broken_heart: [Build #102906](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/) failed db8827776ea554aa0de15fc9c3796b8e605afa6a
* :green_heart: [Build #102525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/) succeeded 064794b7e707dd21cc192839a2de5457ec40df44
* :green_heart: [Build #102128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/) succeeded 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5
* :green_heart: [Build #101422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/) succeeded c7240abe135cda0ea68b10cd460e25cfac2b0fac
* :green_heart: [Build #101142](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/) succeeded 82d7e9f9ce1354bdaceea075591c9b3786b59159

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101142/","number":"101142","commit":"82d7e9f9ce1354bdaceea075591c9b3786b59159"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/","number":"102906","commit":"db8827776ea554aa0de15fc9c3796b8e605afa6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/","number":"102922","commit":"db3ab50aeb27af42cda55f50e8667e11805059b0"}]}
PIPELINE-->

flash1293 commented:
@dej611 @wylieconlon I added an indicator for the dimension trigger and moved the flag into the dimension editor.

With the indicator I think a separate popover is not really justified and the consistency of where functionalities are shown is more important.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/)
* Commit: 4ebca5f6adb8ebac697ff72a9599d399e25a490e
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/execution/node/515/log)

### History
* :green_heart: [Build #102922](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/) succeeded db3ab50aeb27af42cda55f50e8667e11805059b0
* :broken_heart: [Build #102906](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/) failed db8827776ea554aa0de15fc9c3796b8e605afa6a
* :green_heart: [Build #102525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/) succeeded 064794b7e707dd21cc192839a2de5457ec40df44
* :green_heart: [Build #102128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/) succeeded 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5
* :green_heart: [Build #101422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/) succeeded c7240abe135cda0ea68b10cd460e25cfac2b0fac

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101422/","number":"101422","commit":"c7240abe135cda0ea68b10cd460e25cfac2b0fac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/","number":"102906","commit":"db8827776ea554aa0de15fc9c3796b8e605afa6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/","number":"102922","commit":"db3ab50aeb27af42cda55f50e8667e11805059b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/","number":"103389","commit":"4ebca5f6adb8ebac697ff72a9599d399e25a490e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103396/)
* Commit: b2ba6b9b644ad83cf73a245d740128e56460337f
* [Documentation Changes](https://kibana_88680.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e108e5bb2f1ccbdc15a58ae58779ff3fd0d1d8ce) | [after](https://github.com/elastic/kibana/commit/b2ba6b9b644ad83cf73a245d740128e56460337f) | diff |
| --- | --- | --- | --- |
| `lens` | 478 | 479 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e108e5bb2f1ccbdc15a58ae58779ff3fd0d1d8ce) | [after](https://github.com/elastic/kibana/commit/b2ba6b9b644ad83cf73a245d740128e56460337f) | diff |
| --- | --- | --- | --- |
| `lens` | 873.7KB | 876.0KB | +2.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e108e5bb2f1ccbdc15a58ae58779ff3fd0d1d8ce) | [after](https://github.com/elastic/kibana/commit/b2ba6b9b644ad83cf73a245d740128e56460337f) | diff |
| --- | --- | --- | --- |
| `lens` | 34.3KB | 34.2KB | -100.0B |

### History
* :broken_heart: [Build #103389](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/) failed 4ebca5f6adb8ebac697ff72a9599d399e25a490e
* :green_heart: [Build #102922](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/) succeeded db3ab50aeb27af42cda55f50e8667e11805059b0
* :broken_heart: [Build #102906](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/) failed db8827776ea554aa0de15fc9c3796b8e605afa6a
* :green_heart: [Build #102525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/) succeeded 064794b7e707dd21cc192839a2de5457ec40df44
* :green_heart: [Build #102128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/) succeeded 00b7ab667612cfa09e5db3a707c3bf74eab1e6d5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102128/","number":"102128","commit":"00b7ab667612cfa09e5db3a707c3bf74eab1e6d5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/","number":"102906","commit":"db8827776ea554aa0de15fc9c3796b8e605afa6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/","number":"102922","commit":"db3ab50aeb27af42cda55f50e8667e11805059b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/","number":"103389","commit":"4ebca5f6adb8ebac697ff72a9599d399e25a490e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103396/","number":"103396","commit":"b2ba6b9b644ad83cf73a245d740128e56460337f"}]}
PIPELINE-->

dej611 reviewed (changes requested):
The "hidden" icon is not shown for `Metrics`:

<img width="1044" alt="Screenshot 2021-02-03 at 19 47 00" src="https://user-images.githubusercontent.com/924948/106794317-f5af6580-6658-11eb-96ce-c8f45710a3dd.png">

I may also suggest to use the same `eyeClosed` icon for the column dropdown menu for consistency.

<img width="1038" alt="Screenshot 2021-02-03 at 19 48 14" src="https://user-images.githubusercontent.com/924948/106794388-1081da00-6659-11eb-8448-a2fd05789619.png">

I am not fully convinced about the ability to hide all columns and show a blank canvas: perhaps at least one column has to be shown? Not a blocker, but it would be nice to have it sorted.

wylieconlon reviewed (approved):
Tested again in FF and the new UI seems to have no issues with syncing. LGTM!

flash1293 commented:
@elasticmachine merge upstream

flash1293 commented:
@dej611 

> The "hidden" icon is not shown for Metrics:

Thanks, fixed

> I am not fully convinced about the ability to hide all columns and show a blank canvas: perhaps at least one column has to be shown? Not a blocker, but it would be nice to have it sorted.

I don't really see when it would be relevant, but I don't have a strong opinion on it either - added the logic to disable the last visible column switch.

dej611 reviewed (changes requested):
While the panel toggle component is disabled for the last column:

<img width="1060" alt="Screenshot 2021-02-04 at 11 06 07" src="https://user-images.githubusercontent.com/924948/106877633-449cdf80-66d9-11eb-93c9-54b40082c632.png">

You can still have a blank table passing thru the column dropdown `Hide`:

<img width="716" alt="Screenshot 2021-02-04 at 11 06 30" src="https://user-images.githubusercontent.com/924948/106877646-48c8fd00-66d9-11eb-9274-fbd240f646d2.png">

As said, it's not a blocker, either way is fine as long as it's consistent.

Also the `Reset width` action in the column dropdown is always enabled. If clicked nothing happens.

flash1293 commented:
@dej611 Fixed that as well and also added the same icon as you mentioned above.

dej611 reviewed (approved):
All tested on Chrome and FF. 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103756/)
* Commit: 7847e868ad38b5571ed836ed4972d6765cc6a51c
* [Documentation Changes](https://kibana_88680.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/df4eb0f726cbe72c58a84228caa4984764a49cf0) | [after](https://github.com/elastic/kibana/commit/7847e868ad38b5571ed836ed4972d6765cc6a51c) | diff |
| --- | --- | --- | --- |
| `lens` | 478 | 479 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/df4eb0f726cbe72c58a84228caa4984764a49cf0) | [after](https://github.com/elastic/kibana/commit/7847e868ad38b5571ed836ed4972d6765cc6a51c) | diff |
| --- | --- | --- | --- |
| `lens` | 873.7KB | 876.2KB | +2.5KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/df4eb0f726cbe72c58a84228caa4984764a49cf0) | [after](https://github.com/elastic/kibana/commit/7847e868ad38b5571ed836ed4972d6765cc6a51c) | diff |
| --- | --- | --- | --- |
| `lens` | 34.3KB | 34.3KB | -100.0B |

### History
* :green_heart: [Build #103396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103396/) succeeded b2ba6b9b644ad83cf73a245d740128e56460337f
* :broken_heart: [Build #103389](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/) failed 4ebca5f6adb8ebac697ff72a9599d399e25a490e
* :green_heart: [Build #102922](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/) succeeded db3ab50aeb27af42cda55f50e8667e11805059b0
* :broken_heart: [Build #102906](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/) failed db8827776ea554aa0de15fc9c3796b8e605afa6a
* :green_heart: [Build #102525](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/) succeeded 064794b7e707dd21cc192839a2de5457ec40df44

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102525/","number":"102525","commit":"064794b7e707dd21cc192839a2de5457ec40df44"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102906/","number":"102906","commit":"db8827776ea554aa0de15fc9c3796b8e605afa6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102922/","number":"102922","commit":"db3ab50aeb27af42cda55f50e8667e11805059b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103389/","number":"103389","commit":"4ebca5f6adb8ebac697ff72a9599d399e25a490e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103396/","number":"103396","commit":"b2ba6b9b644ad83cf73a245d740128e56460337f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103756/","number":"103756","commit":"7847e868ad38b5571ed836ed4972d6765cc6a51c"}]}
PIPELINE-->
